File: kattis/Bus Numbers.py
#n = int(input())
m = sorted(map(int, input().split()))
m.append(9999)

res = []  # answer
current = m[0]  # placeholder
counter = 0  # counts the number of consecutives


for i in range(len(m)-1):
    if m[i] == (m[i+1]-1):
        counter += 1
        current = m[i+1 - counter]
    elif m[i] != (m[i+1]-1) and counter == 0:
        res.append(m[i])
    else:
        res.append(str(current) + "-" + str(m[i]))
        counter = 0

if counter != len(m)-1 and m[-1] not in res:
    res.append(m[-1])

if counter == len(m)-1:
    res.append(str(m[0]) + "-" + str(m[-1]))

res.pop()
print(" ".join(map(str, res)))

File: kattis/busNumbers.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
});

let first = true;
let res = [];
rl.on("line", (line) => {
  if (first) {
    n = +line;
    first = false;
    if (!n || n == 0) {
      rl.close(); // if no input, or falsy input end the program
      return;
    }
    return; // continue to next line read by returning to top of line
  }
  let busNumbers = line
    .split(" ")
    .map(Number)
    .sort((a, b) => a - b); // take numbers
  res.push(busNumbers[0]);
  for (i = 1; i < busNumbers.length - 1; i++) {
    if (
      busNumbers[i - 1] == busNumbers[i] - 1 &&
      busNumbers[i + 1] == busNumbers[i] + 1
    ) {
      if (res[res.length - 1] == "-") continue;
      res.push("-");
    } else {
      if (res[res.length - 1] != "-") res.push(" ");
      res.push(busNumbers[i]);
    }
  }
  if (res[res.length - 1] != "-") res.push(" ");
  if (n != 1) res.push(busNumbers[busNumbers.length - 1]);
  console.log(res.join(""));
  rl.close();
});
